cmd/traffic/cmd/manager/internal/mutator/watcher.go

```diff
@@ -389,66 +399,123 @@ func (c *configWatcher) Store(ctx context.Context, ac *agentconfig.Sidecar, upda
 		return nil
 	}
 
-	create := false
-	api := k8sapi.GetK8sInterface(ctx).CoreV1().ConfigMaps(ns)
-	cm, err := api.Get(ctx, agentconfig.ConfigMap, meta.GetOptions{})
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			return fmt.Errorf("unable to get ConfigMap %s: %w", agentconfig.ConfigMap, err)
+	c.configUpdatersMapLock.Lock()
+	var cu *configUpdater
+
+	newGroup := false
+	if _, ok := c.configUpdaters[ns]; !ok {
+		grp, _ := errgroup.WithContext(ctx)
+		c.configUpdaters[ns] = &configUpdater{
+			grp,
```

Comment:
I would include the `ns` and `ctx` in the `configUpdater` struct and add them here. That way, the `updateConfigMap` no longer needs any arguments, which in turn means that it can be used directly as `cu.Go(c.updateConfigMap)` on line 426 instead of returning a no-args function. It would also address my comment about what context that gets passed on to that function.